Userland/Utilities/cat.cpp

```diff
@@ -37,13 +41,46 @@ ErrorOr<int> serenity_main(Main::Arguments arguments)
 
     TRY(Core::System::pledge("stdio"));
 
+    size_t line_count = 1;
     Array<u8, 32768> buffer;
     for (auto const& file : files) {
         while (!file->is_eof()) {
             auto const buffer_span = TRY(file->read_some(buffer));
-            out("{:s}", buffer_span);
+            if (show_lines) {
+                line_count = output_buffer_with_line_numbers(line_count, buffer_span);
+            } else {
+                out("{:s}", buffer_span);
+            }
         }
+        line_count = 1;
```

Comment:
GNU's `cat` doesn't seem to reset the counter between files.

Reply:
Good point, I have updated to match.